File: uk-ets-app-registry/uk-ets-registry-api/src/main/java/gov/uk/ets/registry/api/user/KeycloakUser.java
package gov.uk.ets.registry.api.user;

import java.util.List;
import java.util.Map;
import java.util.Set;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.keycloak.representations.idm.UserRepresentation;

@Getter
@Setter
@NoArgsConstructor
public class KeycloakUser {

    private String username;
    private String email;
    private String firstName;
    private String lastName;
    private boolean eligibleForSpecificActions;
    private Map<String, List<String>> attributes;
    private Set<String> userRoles;
    /**
     *  The KeycloakUser constructor.
     * @param userRepresentation the UserRepresentation object.
     */
    public KeycloakUser(UserRepresentation userRepresentation) {
        this.username = userRepresentation.getUsername();
        this.email = userRepresentation.getEmail();
        this.firstName = userRepresentation.getFirstName();
        this.lastName = userRepresentation.getLastName();
        this.attributes = userRepresentation.getAttributes();
    }
}

File: uk-ets-app-registry/uk-ets-registry-web/e2e/src/util/test-data/registry-db/user/model/keycloak-user.ts
import { KeycloakUserBuilder } from './keycloak-user.builder';

export class KeycloakUser {
  username: string;
  enabled = true;
  email: string;
  firstName: string;
  lastName: string;
  emailVerified = true;
  attributes: {
    urid: string[];
    alsoKnownAs: string[];
    buildingAndStreet: string[];
    buildingAndStreetOptional: string[];
    buildingAndStreetOptional2: string[];
    country: string[];
    countryOfBirth: string[];
    postCode: string[];
    townOrCity: string[];
    stateOrProvince: string[];
    yearOfBirth: string[];
    state: string[];
    workBuildingAndStreet: string[];
    workBuildingAndStreetOptional: string[];
    workBuildingAndStreetOptional2: string[];
    workCountry: string[];
    workCountryCode: string[];
    workPhoneNumber: string[];
    workPostCode: string[];
    workTownOrCity: string[];
    workStateOrProvince: string[];
  };
  credentials: [
    { type: string; value: string },
    {
      type: string;
      secretData: string;
      temporary: boolean;
      credentialData: string;
    },
  ];

  realmRoles: string[];
  clientRoles: {
    account: string[];
    'uk-ets-registry-api': string[];
  };

  constructor(builder: KeycloakUserBuilder) {
    this.username = builder.getUsername;
    this.email = builder.getEmail;
    this.firstName = builder.getFirstName;
    this.lastName = builder.getLastName;
    this.attributes = builder.getAttributes;
    this.credentials = builder.getCredentials;
    this.realmRoles = builder.realmRoles;
    this.clientRoles = builder.getClientRoles;
  }
}
